fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java

```diff
@@ -72,7 +73,13 @@ public Expression visitNot(Not not, CascadesContext context) {
     @Override
     public Expression visitExistsSubquery(Exists exists, CascadesContext context) {
         AnalyzedResult analyzedResult = analyzeSubquery(exists);
-
+        if (analyzedResult.rootIsLimitZero()) {
+            return BooleanLiteral.of(exists.isNot());
+        }
+        if (analyzedResult.isCorrelated() && analyzedResult.rootIsLimitWithOffset()) {
+            throw new AnalysisException("Unsupported correlated subquery with a LIMIT clause with offset > 0 "
```

Comment:
select * from t1 where not exists (select * from t2 where t1.c1 = t2.c1 limit 0 offset 1);
Means doris will throw exception ? I think it makes sence to return full set of t1.
and for exists, it returns empty sets also.

Reply:
ignore above cases.
For this case, select * from t1 where [not] exists (select * from t2 where t1.c1 = t2.c1 limit 1 offset 1);
will doris throw exception?

Reply:
currently select * from t1 where [not] exists (select * from t2 where t1.c1 = t2.c1 limit 1 offset 1); will return wrong result.